test/unit/helpers/peersManager.js

```diff
@@ -0,0 +1,246 @@
+'use strict';
+
+var _ = require('lodash');
+var chai = require('chai');
+var expect = require('chai').expect;
+var sinon = require('sinon');
+var randomPeer = require('../../common/objectStubs').randomPeer;
+var Peer = require('../../../logic/peer');
+
+var peersManager = require('../../../helpers/peersManager');
+
+describe('PeersManager', function () {
+
+	var validPeer;
+
+	beforeEach(function () {
+		validPeer = new Peer(randomPeer);
+		peersManager.peers = {};
+		peersManager.addressToNonceMap = {};
+		peersManager.nonceToAddressMap = {};
+	});
+
+	describe('constructor', function () {
+
+		it('should have empty peers map after initialization', function () {
+			expect(peersManager).to.have.property('peers').to.be.empty;
+		});
+
+		it('should have empty addressToNonceMap map after initialization', function () {
+			expect(peersManager).to.have.property('addressToNonceMap').to.be.empty;
+		});
+
+		it('should have empty nonceToConnectionIdMap map after initialization', function () {
+			expect(peersManager).to.have.property('nonceToAddressMap').to.be.empty;
+		});
+	});
+
+	describe('add', function () {
+
+		it('should return false when invoked without arguments', function () {
+			expect(peersManager.add()).not.to.be.ok;
+		});
+
+		it('should return false when invoked with peer equal null', function () {
+			expect(peersManager.add(null)).not.to.be.ok;
+		});
+
+		it('should return false when invoked with peer equal 0', function () {
+			expect(peersManager.add(0)).not.to.be.ok;
+		});
+
+		it('should return false when invoked with peer has no string field', function () {
+			expect(peersManager.add({})).not.to.be.ok;
+		});
+
+		it('should add entry to peers when invoked with valid arguments', function () {
+			peersManager.add(validPeer);
+			expect(peersManager.peers).to.have.property(validPeer.string).eql(validPeer);
+		});
+
+		it('should add entry to addressToNonceMap when invoked with valid arguments', function () {
+			peersManager.add(validPeer);
+			expect(peersManager.addressToNonceMap).to.have.property(validPeer.string).eql(validPeer.nonce);
+		});
+
+		it('should add entry to nonceToAddressMap when invoked with valid arguments', function () {
+			peersManager.add(validPeer);
+			expect(peersManager.nonceToAddressMap).to.have.property(validPeer.nonce).eql(validPeer.string);
+		});
+
+		describe('multiple valid entries', function () {
+
+			var validPeerA, validPeerB;
+
+			beforeEach(function () {
+
+				validPeerA = _.clone(validPeer);
+				validPeerA.string += 'A';
+				validPeerA.nonce += 'A';
+
+				validPeerB = _.clone(validPeer);
+				validPeerB.string += 'B';
+				validPeerB.nonce += 'B';
+
+				expect(peersManager.add(validPeerA)).to.be.ok;
+				expect(peersManager.add(validPeerB)).to.be.ok;
+			});
+
+			it('should add multiple entries in peers after multiple valid entries added', function () {
```

Comment:
should contain multiple entries in peers after multiple valid entries added.

same for the next two.